### DISCUSSION TITLE
Use a variable from another state ?

### DISCUSSION THREAD

**Discussion by NalegProto** [HaxeFlixel/flixel#2360] [Q&A] [answered]
Hey ! I need some help with my code.

I'm trying to make a shop using a `ShopState` and the `PlayState`.
With the PlayState, I access the ShopState, buy the item, and this action updates a boolean from the PlayState.
Problem is, I don't find any way to use the boolean from another state.

Can I get help on this ?

**Comment by RichardBray** [comment, marked as answer]
Might be helpful to look at some code but you have several options here:

1 - Make the boolean from the other state public so you can access it in another class
2 - Create a singleton and add the information you want to share between states in there so different states can access it
3 - Put the info you want to share in the games save state and when you load the game pass the loaded saved data into each state
4 - Create a public function that will return the boolean value and call it in another class

I personally use the second option for most of my games.

**Reply by NalegProto** [reply]
I'm using the first option

ShopState :
```haxe
package;

//imports, useless at this point

class ShopState extends FlxState
{
     public var isClicked = false;
     // Rest of the code, useless at this point
}
```

PlayState :
```haxe
package;

import flixel.text.FlxText;

class PlayState extends FlxState
{
     override public function create()
     {
          var text:FlxText;
          text = new FlxText(100, 100, 0, "isClicked variable is set as " + Std.string(ShopState.isClicked), 16);
     }
     //Rest of the code, useless at this point
}
```

I have an error when using the variable in the FlxText of PlayState, saying `Class <ShopState> has no field isClicked`

EDIT: I found out how to use it by replacing `public var isClicked = false` by `public static var isClicked = false`

**Reply by RichardBray** [reply]
Nice, glad you were able to figure it out. Yeah if you don't want to instantiate the class then adding the `static` makes sense.
